sycl/include/CL/sycl/define_vendors.hpp

```diff
@@ -0,0 +1,13 @@
+//==---------- define_vendors.hpp ----- Preprocessor directives ------------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+#define SYCL_IMPLEMENTATION_ONEAPI
```

Comment:
Could you please clarify why we define this macro?

Reply:
@romanovvlad https://www.khronos.org/registry/SYCL/specs/sycl-2020/html/sycl-2020.html#_preprocessor_directives_and_macros

Reply:
Do not see anything about defining `SYCL_IMPLEMENTATION_ONEAPI` in section 5.6.
Could you please copy paste specific wording which requires us to define this macro?

Reply:
@romanovvlad sorry, I've forgot to put the link the first time I posted the comment. I've updated it with the link to  section 6.2, which reads:

> The implementation must predefine at least one macro of the form SYCL_IMPLEMENTATION_<vendorstring> which allows applications to test whether they are being compiled with that vendor’s implementation. For example, the Acme vendor could predefine a macro whose name is SYCL_IMPLEMENTATION_ACME.

Reply:
Thanks, it sounds like `SYCL_IMPLEMENTATION_INTEL` covers that. Could you please clarify why `SYCL_IMPLEMENTATION_ONEAPI` is being added?

Reply:
Discussed offline.

Reply:
We provide extensions in both `ext::oneapi` and `ext::intel` namespaces. My understanding is that we must provide both macros as well. Tagging @gmlueck

Reply:
Agreed, we should define both.